flytekit/common/tasks/notebook.py

```diff
@@ -0,0 +1,542 @@
+from __future__ import absolute_import
+
+import flytekit.common.interface as _interface
+from flytekit.models.interface import Variable
+import os as _os
+from flytekit.common.tasks import output as _task_output
+import json
+import papermill as pm
+from flytekit import __version__
+import importlib as _importlib
+
+import datetime as _datetime
+from pyspark import SparkConf, SparkContext
+import sys as _sys
+import six as _six
+from flytekit.bin import spark_executor
+from flytekit.common import constants as _constants
+from flytekit.common.exceptions import scopes as _exception_scopes
+from flytekit.common.tasks import sdk_runnable as _sdk_runnable
+from flytekit.common.tasks import spark_task as _spark_task
+
+from flytekit.common.types import helpers as _type_helpers
+from flytekit.models import literals as _literal_models, task as _task_models
+from google.protobuf.json_format import MessageToDict as _MessageToDict
+from flytekit.sdk.types import Types
+
+from flytekit.common.types.helpers import pack_python_std_map_to_literal_map as _packer
+from flytekit.common.types import primitives as _p
+from google.protobuf import text_format
+from flytekit.common import sdk_bases
+from flytekit.common.tasks import task as base_tasks
+from flytekit.engines import loader as _engine_loader
+
+type_map = {
+    int: _p.Integer,
+    bool: _p.Boolean,
+    float: _p.Float,
+    str: _p.String,
+    _datetime.datetime: _p.Datetime,
+    _datetime.timedelta: _p.Timedelta,
+}
+
+OUTPUT_NOTEBOOK = 'output_notebook'
+
+def convert_outputs(outputs=None):
+    """
+    outputs: dict
+    Records the output
+    """
+    if outputs is None:
+        return _packer({}, {})
+    tm = {}
+    for k, v in _six.iteritems(outputs):
+        t = type(v)
+        if t not in type_map:
+            raise ValueError(
+                "Currently only primitive types {} are supported for recording from notebook".format(type_map))
+        tm[k] = type_map[t]
+    return _packer(outputs, tm)
+
+
+def record_outputs(outputs=None):
+    p = convert_outputs(outputs)
+    return p.to_flyte_idl()
+
+# TODO: Support Client Mode
+def get_spark_context(spark_conf):
```

Comment:
Should this be upstreamed to spark task? Wouldn't that suffer from the same issue.

Reply:
no, this is only for notebook_spark_tasks